tmt/steps/provision/podman.py

```diff
@@ -128,16 +128,18 @@ def start(self):
                 ['pull', '-q', self.image],
                 message=f"Pull image '{self.image}'.")
 
+        # Mount the whole run directory in the container
+        rundir = self.parent.plan.run.workdir
+
         # Deduce container name from run id, as it can be a path,
         # make it podman container name friendly
-        tmt_workdir = self.parent.plan.workdir
-        self.container = 'tmt' + tmt_workdir.replace('/', '-')
+        self.container = 'tmt' + rundir.replace('/', '-')
         self.verbose('name', self.container, 'green')
 
         # Run the container
         self.container_id = self.podman(
             ['run'] + ['--name', self.container,
-            '-v', f'{tmt_workdir}:{tmt_workdir}:Z', '-itd', self.image],
+            '-v', f'{rundir}:{rundir}:Z', '-itd', self.image],
```

Comment:
when I'm seeing this under podman, that theoretically you mount folder inside podman, and do not use `podman cp` and pull/push operations, maybe there may exist another similar optimizations for virtual machines, e.g.
 - use `sshfs` (probably has to be called on guest) this could be bad. 
 - use `libguestfs` tools to work with VM images and copy files directly from/to image not do it via `ssh(rsync)`
 - map VM discs to local machine e.g. `loopback` device and then mount the discs locally (not sure if possible without sudo)
 - unify copy operation on local containers (VMs) `machinectl copy-to/from`

Bad is, that various types needs various handling how to copy/share files across systems, but it is also current situation.

Take it as an brainstorming, what I'm thinking about :-)

Reply:
@jscotka thanks for the suggestion. 

> use sshfs (probably has to be called on guest) this could be bad.

I have not good experience with this, I would rather avoid it

> use libguestfs tools to work with VM images and copy files directly from/to image not do it via ssh(rsync)

This could work with testcloud, but it will not with minute provisioner. I would rather stick to `rsync` which works in all cases of VM provisioning

> unify copy operation on local containers (VMs) machinectl copy-to/from

I see the same problem as with the one before.

> Bad is, that various types needs various handling how to copy/share files across systems, but it is also current situation.

Well sticking to `rsync` with all VMs and volume mounting to containers, does not seem so bad to me.

> Take it as an brainstorming, what I'm thinking about :-)

I did, thanks for the suggestions, it is good to have this conversation. 
￼

Reply:
@jscotka if you want to file for this an issue, if you think there could be an improvement, pls file a separate issue / RFE. I consider this resolved for this PR.